### PR TITLE
Implement SendHapticData RPC

### PR DESCRIPTION
**Pull request by masatoogawa**
sdl_evolution SDL-0075 is implemented.

SDL Core PR is [here](https://github.com/smartdevicelink/sdl_core/pull/1712)

Please review them.

### CODE REVIEW

**Comment by masatoogawa** [comment]
> Should be ATF_SendHapticData.lua , not ATF_HapticRequest.lua

fixed at 9a5aa90c93ef7a9ea8a70cc0b2811a46cdc07d03

**Comment by masatoogawa** [comment]
With the acceptance of https://github.com/smartdevicelink/sdl_evolution/pull/254, I will fix the tests.

**Comment by masatoogawa** [comment]
> With the acceptance of https://github.com/smartdevicelink/sdl_evolution/pull/254, I will fix the tests.

fixed at https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1622/commits/45089990d5aae80f60a2e6cbc9a627814fa64e65

File: test_scripts/API/ATF_HapticRequest.lua [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+Test = require('connecttest')
+local commonFunctions =
+  require('user_modules/shared_testcases/commonFunctions')
+
+--------------------------------------------------------------------------------
+-- TEST BLOCK I : Check normal cases
+--------------------------------------------------------------------------------
+commonFunctions:newTestCasesGroup("Test suite: Check normal cases")
+
```
Comment (jacobkeeler) [inline test_scripts/API/ATF_HapticRequest.lua]: Making this test pass requires changing the preloaded PT manually to allow `SendHapticData`, there is a way to do this automatically with ATF, use the [Alert](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/master/test_scripts/API/ATF_Alert.lua#L101) test cases as an example

File: test_scripts/API/ATF_SendHapticData.lua [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+Test = require('connecttest')
+local commonFunctions =
+  require('user_modules/shared_testcases/commonFunctions')
+
+--------------------------------------------------------------------------------
+-- TEST BLOCK I : Check normal cases
+--------------------------------------------------------------------------------
+commonFunctions:newTestCasesGroup("Test suite: Check normal cases")
+
```
Comment (jacobkeeler) [inline test_scripts/API/ATF_SendHapticData.lua]: Making this test pass requires changing the preloaded PT manually to allow `SendHapticData`, there is a way to do this automatically with ATF, use the [Alert](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/master/test_scripts/API/ATF_Alert.lua#L101) test cases as an example

File: test_scripts/API/ATF_HapticRequest.lua [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    "UI.SendHapticData",
+    {
+      HapticSpatialData =
+      { { id = 1, x = 2.0, y = 3.0, width = 4.0, height = 5.0, } }
+    }
+  )
+  :Do(function(_,data)
+    self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  -- mobiel side
```
Comment (jacobkeeler) [inline test_scripts/API/ATF_HapticRequest.lua]: Not relevant anymore

File: test_scripts/API/ATF_SendHapticData.lua (hunk 2 above)
Comment (masatoogawa) [reply test_scripts/API/ATF_SendHapticData.lua]: @jacobkeeler Thank you for your review. Fixed at 9158befab982085f9e0016f6b410711e5be6aa06

**Pull request merged**